### PR TITLE
Refactoring pundit for authentication

### LINKED ISSUE #10083: Replace `require_login` controller filters with Pundit policies
Some controllers rely solely on Pundit and others on custom legacy code like [require_login](https://github.com/openSUSE/open-build-service/blob/92b16b72e57e483f7085441a2a39c5864c4ed28f/src/api/app/lib/authenticator.rb#L69-L71) to authorize various actions throughout the application.

We want to rely completely on Pundit to use a widely adopted authorization system instead of our own custom solution.

### PR DESCRIPTION
**Pull request by hennevogel**
In #10083 we extended our pundit `ApplicationPolicy`  with an option to ensure that someone is logged in. We then started to bake this into controllers. I think the analysis of #10083 is wrong, Pundit is for authorization a.k.a. who can do what. `require_login` is for ensuring someone is authenticated a.k.a `User.session` is set.

It should have been clear that this isn't the best idea by the amount of custom code we had to add to pundit policies and controllers. Let's roll this back.

Closes #10083

### CODE REVIEW

**Comment by hennevogel** [comment]
I'm not going to touch the codeclimate complaint in `check_creator`. I have no idea what this fixes and what the logic is supposed to be. I just moved code...

File: src/api/app/controllers/person/token_controller.rb [hunk 1]
```diff
@@ -5,12 +5,13 @@ class TokenController < ApplicationController
 
     before_action :set_user
     before_action :validate_operation, only: [:create]
+    after_action :verify_authorized
 
     # GET /person/<login>/token
     def index
-      authorize @user, :show?
+      authorize @user, :update?
```
Comment (danidoni) [inline src/api/app/controllers/person/token_controller.rb:12]: Why `update?` if its a read?

Comment (hennevogel) [reply src/api/app/controllers/person/token_controller.rb:12]: Not sure what you mean. The name of the policy method?

Comment (danidoni) [reply src/api/app/controllers/person/token_controller.rb:12]: Yes

Comment (hennevogel) [reply src/api/app/controllers/person/token_controller.rb:12]: Because you should only be able to see the tokens if you can change them.

Comment (danidoni) [reply src/api/app/controllers/person/token_controller.rb:12]: Makes sense

**Review by danidoni — APPROVED** [review]

**Pull request merged**